kernel/src/hil/time.rs

```diff
@@ -175,3 +177,53 @@ pub trait TimerClient {
     /// Callback signaled when the timer's clock reaches the specified interval.
     fn fired(&self);
 }
+
+/// The `TimerWrappingOps` trait is used to abstract the timer
+/// addition and subtraction operations in order to support timers of
+/// different bit lengths. Currently implementations exists for 32-bit
+/// and 24-bit timers.
+pub trait TimerWrappingOps {
+    /// Add timer tics
+    ///
+    /// For 24-bit timers, the `x` and `y` values cannot exceed `2^24
+    /// - 1`.
+    fn wrapping_add(x: u32, y: u32) -> u32;
+
+    /// Subtract timer tics
+    ///
+    /// For 24-bit timers, the `x` and `y` values cannot exceed `2^24
+    /// - 1`.
+    fn wrapping_sub(x: u32, y: u32) -> u32;
+}
+
+/// 32-bit `TimerWrappingOps`
+#[derive(Debug)]
+pub struct Timer32Bit;
+impl TimerWrappingOps for Timer32Bit {
+    fn wrapping_add(x: u32, y: u32) -> u32 {
+        x.wrapping_add(y)
+    }
+
+    fn wrapping_sub(x: u32, y: u32) -> u32 {
+        x.wrapping_sub(y)
+    }
+}
+
+/// 24-bit `TimerWrappingOps`
+#[derive(Debug)]
+pub struct Timer24Bit;
+impl TimerWrappingOps for Timer24Bit {
+    fn wrapping_add(x: u32, y: u32) -> u32 {
+        let max_value = (1 << 24) - 1;
+
+        let res = x + y;
+
+        res & max_value
```

Comment:
Why not putting the whole expression on the same line, as for subtraction? Also, even if overflow isn't expected, I think `x.wrapping_add(y) & max_value` would be clearer, for consistency with subtraction.